experimental/wbuild-examples/FWF/src/XmIcon/XmIcon.man
'\" t
.TH "" 3 "" "Version 3.0" "Free Widget Foundation"
.SH NAME
XmIcon
.SH DESCRIPTION
The XmIcon widget displays an image. The preferred width and height
of the widget will be set to the width and height of the loaded
image. The widget has a callback \fIactivateCallback\fP, that is by
default bound to a click of the left mouse button.

The image is a \fIPixmap\fP, optionally accompanied by a mask, to make
parts of the image transparent.

.SS "Public variables"

.ps -2
.TS
center box;
cBsss
lB|lB|lB|lB
l|l|l|l.
XmIcon
Name	Class	Type	Default
XtNimage	XtCImage	Icon *	NULL 
XtNactivate	XtCActivate	Callback	NULL 

.TE
.ps +2

.TP
.I "XtNimage"
The image must be in \fIPixmap\fP format. The width and height of the
image will be used to set the width and height of the widget.  There
is a converter for strings, which will
try to interpret a string as a file name and load the file. The file
must be in XPM format. The converter also knows about some built-in
images, called \fI"FATAL"\fP, \fI"ERROR"\fP, \fI"WARNING"\fP, \fI"QUESTION"\fP,
\fI"INFO"\fP and \fI"NONE"\fP.

	

.TP
.I "XtNactivate"
The callback is called by the \fIactivate\fP action, which is by default
bound to a click of mouse button 1.

	

.ps -2
.TS
center box;
cBsss
lB|lB|lB|lB
l|l|l|l.
XmManager
Name	Class	Type	Default
XtNbottom_shadow_color	XtCBottom_shadow_color	Pixel 	0 
XtNbottom_shadow_pixmap	XtCBottom_shadow_pixmap	Pixmap 	0 
XtNforeground	XtCForeground	Pixel 	0 
XtNhelp_callback	XtCHelp_callback	Callback	NULL 
XtNhighlight_color	XtCHighlight_color	Pixel 	0 
XtNhighlight_pixmap	XtCHighlight_pixmap	Pixmap 	None 
XtNinitial_focus	XtCInitial_focus	Widget 	NULL 
navigationType	XtCNavigationType	NavigationType	XmTAB_GROUP 
XtNshadow_thickness	XtCShadow_thickness	Dimension 	0 
XtNstring_direction	XtCString_direction	XMStringDirection 	0 
XtNtop_shadow_color	XtCTop_shadow_color	Pixel 	0 
XtNtop_shadow_pixmap	XtCTop_shadow_pixmap	Pixmap 	None 
XtNtraversal_on	XtCTraversal_on	Boolean 	True 
XtNunit_type	XtCUnit_type	Unsigned  char 	XmPIXELS 
XtNuser_data	XtCUser_data	XTPointer 	NULL 

.TE
.ps +2

.ps -2
.TS
center box;
cBsss
lB|lB|lB|lB
l|l|l|l.
Composite
Name	Class	Type	Default
XtNchildren	XtCChildren	WidgetList 	NULL 
insertPosition	XtCInsertPosition	XTOrderProc 	NULL 
numChildren	XtCNumChildren	Cardinal 	0 

.TE
.ps +2

.ps -2
.TS
center box;
cBsss
lB|lB|lB|lB
l|l|l|l.
Core
Name	Class	Type	Default
XtNx	XtCX	Position 	0 
XtNy	XtCY	Position 	0 
XtNwidth	XtCWidth	Dimension 	0 
XtNheight	XtCHeight	Dimension 	0 
borderWidth	XtCBorderWidth	Dimension 	0 
XtNcolormap	XtCColormap	Colormap 	NULL 
XtNdepth	XtCDepth	Int 	0 
destroyCallback	XtCDestroyCallback	XTCallbackList 	NULL 
XtNsensitive	XtCSensitive	Boolean 	True 
XtNtm	XtCTm	XTTMRec 	NULL 
ancestorSensitive	XtCAncestorSensitive	Boolean 	False 
accelerators	XtCAccelerators	XTTranslations 	NULL 
borderColor	XtCBorderColor	Pixel 	0 
borderPixmap	XtCBorderPixmap	Pixmap 	NULL 
background	XtCBackground	Pixel 	0 
backgroundPixmap	XtCBackgroundPixmap	Pixmap 	NULL 
mappedWhenManaged	XtCMappedWhenManaged	Boolean 	True 
XtNscreen	XtCScreen	Screen *	NULL 

.TE
.ps +2

.SS "Exports"

The type \fIIcon\fP is defined in \fIConverters.h\fP

.nf

.B incl
 <Xfwf/Converters.h>
.fi

.SS "Translations"

By default, the \fIactivate\fP action is bound to a mouse click and to
the Return key.

	

.nf
<Btn1Down>,<Btn1Up>: activate() 
.fi

.nf
<Key>Return: activate() 
.fi

.SS "Actions"

.TP
.I "activate

The \fIactivate\fP action just calls the \fIactivate\fP callback functions,
passing the \fIXEvent\fP pointer in the \fIcall_data\fP argument.



wbuild-examples/FWF/src/XmIcon/XmIcon.c
/* Generated by wbuild from "XmIcon.w"
** (generator version $Revision: 3.0 $ of $Date: 95/08/30 17:50:38 $)
*/
#include <X11/IntrinsicP.h>
#include <X11/StringDefs.h>
#line 142 "XmIcon.w"
#include <X11/extensions/shape.h>
#line 143 "XmIcon.w"
#include <stdio.h>
#line 144 "XmIcon.w"
#include <X11/xpm.h>
#include <Xfwf/XmIconP.h>
#line 135 "XmIcon.w"
static void activate(
#if NeedFunctionPrototypes
Widget,XEvent*,String*,Cardinal*
#endif
);

static XtActionsRec actionsList[] = {
{"activate", activate},
};

static char defaultTranslations[] = "\
<Btn1Down>,<Btn1Up>: activate() \n\
<Key>Return: activate() \n\
";
static void _resolve_inheritance(
#if NeedFunctionPrototypes
WidgetClass
#endif
);
#line 53 "XmIcon.w"
static void class_initialize(
#if NeedFunctionPrototypes
void
#endif
);
#line 66 "XmIcon.w"
static void initialize(
#if NeedFunctionPrototypes
Widget ,Widget,ArgList ,Cardinal *
#endif
);
#line 79 "XmIcon.w"
static Boolean  set_values(
#if NeedFunctionPrototypes
Widget ,Widget ,Widget,ArgList ,Cardinal *
#endif
);
#line 102 "XmIcon.w"
static void realize(
#if NeedFunctionPrototypes
Widget,XtValueMask *,XSetWindowAttributes *
#endif
);
#line 112 "XmIcon.w"
static void destroy(
#if NeedFunctionPrototypes
Widget
#endif
);

static XtResource resources[] = {
#line 26 "XmIcon.w"
{XtNimage,XtCImage,XtRIcon,sizeof(((XmIconRec*)NULL)->xmIcon.image),XtOffsetOf(XmIconRec,xmIcon.image),XtRImmediate,(XtPointer)NULL },
#line 31 "XmIcon.w"
{XtNactivate,XtCActivate,XtRCallback,sizeof(((XmIconRec*)NULL)->xmIcon.activate),XtOffsetOf(XmIconRec,xmIcon.activate),XtRImmediate,(XtPointer)NULL },
};

XmIconClassRec xmIconClassRec = {
{ /* core_class part */
/* superclass   	*/  (WidgetClass) &xmManagerClassRec,
/* class_name   	*/  "XmIcon",
/* widget_size  	*/  sizeof(XmIconRec),
/* class_initialize 	*/  class_initialize,
/* class_part_initialize*/  _resolve_inheritance,
/* class_inited 	*/  FALSE,
/* initialize   	*/  initialize,
/* initialize_hook 	*/  NULL,
/* realize      	*/  realize,
/* actions      	*/  actionsList,
/* num_actions  	*/  1,
/* resources    	*/  resources,
/* num_resources 	*/  2,
/* xrm_class    	*/  NULLQUARK,
/* compres_motion 	*/  True ,
/* compress_exposure 	*/  XtExposeCompressMultiple ,
/* compress_enterleave 	*/  True ,
/* visible_interest 	*/  False ,
/* destroy      	*/  destroy,
/* resize       	*/  XtInheritResize,
/* expose       	*/  XtInheritExpose,
/* set_values   	*/  set_values,
/* set_values_hook 	*/  NULL,
/* set_values_almost 	*/  XtInheritSetValuesAlmost,
/* get_values+hook 	*/  NULL,
/* accept_focus 	*/  XtInheritAcceptFocus,
/* version      	*/  XtVersion,
/* callback_private 	*/  NULL,
/* tm_table      	*/  defaultTranslations,
/* query_geometry 	*/  XtInheritQueryGeometry,
/* display_acceleator 	*/  XtInheritDisplayAccelerator,
/* extension    	*/  NULL 
},
{ /* composite_class part */
XtInheritGeometryManager,
XtInheritChangeManaged,
XtInheritInsertChild,
XtInheritDeleteChild,
NULL
},
{ /* constraint_class part */
/* constraint_resources     */  NULL,
/* num_constraint_resources */  0,
/* constraint_size          */  sizeof(XmIconConstraintRec),
/* constraint_initialize    */  NULL,
/* constraint_destroy       */  NULL,
/* constraint_set_values    */  NULL,
/* constraint_extension     */  NULL 
},
{ /* XmManager class part */
#define manager_extension extension
/* translations                 */  XtInheritTranslations ,
/* syn_resources                */  NULL ,
/* num_syn_resources            */  0 ,
/* syn_constraint_resources     */  NULL ,
/* num_syn_constraint_resources */  0 ,
/* parent_process               */  XmInheritParentProcess,
/* manager_extension            */  NULL ,
},
{ /* XmIcon_class part */
 /* dummy */  0
},
};
WidgetClass xmIconWidgetClass = (WidgetClass) &xmIconClassRec;
/*ARGSUSED*/
#line 135 "XmIcon.w"
static void activate(self,event,params,num_params)Widget self;XEvent*event;String*params;Cardinal*num_params;
{
    XtCallCallbackList(self, ((XmIconWidget)self)->xmIcon.activate, event);
}

static void _resolve_inheritance(class)
WidgetClass class;
{
  XmIconWidgetClass c = (XmIconWidgetClass) class;
  XmIconWidgetClass super;
  static CompositeClassExtensionRec extension_rec = {
    NULL, NULLQUARK, XtCompositeExtensionVersion,
    sizeof(CompositeClassExtensionRec), True};
  CompositeClassExtensionRec *ext;
  ext = (XtPointer)XtMalloc(sizeof(*ext));
  *ext = extension_rec;
  ext->next_extension = c->composite_class.extension;
  c->composite_class.extension = ext;
  if (class == xmIconWidgetClass) return;
  super = (XmIconWidgetClass)class->core_class.superclass;
}
#line 53 "XmIcon.w"
/*ARGSUSED*/
#if NeedFunctionPrototypes
#line 53 "XmIcon.w"
static void class_initialize(void)
#else
#line 53 "XmIcon.w"
static void class_initialize()
#endif
#line 54 "XmIcon.w"
{
    static XtConvertArgRec args[] = {
    { XtWidgetBaseOffset, 0, sizeof(Widget) } };

    XtSetTypeConverter(XtRString, "Icon", cvtStringToIcon,
		       args, XtNumber(args), XtCacheNone, NULL);
}
#line 66 "XmIcon.w"
/*ARGSUSED*/
#if NeedFunctionPrototypes
#line 66 "XmIcon.w"
static void initialize(Widget  request,Widget self,ArgList  args,Cardinal * num_args)
#else
#line 66 "XmIcon.w"
static void initialize(request,self,args,num_args)Widget  request;Widget self;ArgList  args;Cardinal * num_args;
#endif
#line 67 "XmIcon.w"
{
    if (((XmIconWidget)self)->xmIcon.image)
	XtVaSetValues
	    (self,
	     XtNwidth, ((XmIconWidget)self)->xmIcon.image->attributes.width,
	     XtNheight, ((XmIconWidget)self)->xmIcon.image->attributes.height,
	     XtNbackgroundPixmap, ((XmIconWidget)self)->xmIcon.image->pixmap,
	     NULL);
}
#line 79 "XmIcon.w"
/*ARGSUSED*/
#if NeedFunctionPrototypes
#line 79 "XmIcon.w"
static Boolean  set_values(Widget  old,Widget  request,Widget self,ArgList  args,Cardinal * num_args)
#else
#line 79 "XmIcon.w"
static Boolean  set_values(old,request,self,args,num_args)Widget  old;Widget  request;Widget self;ArgList  args;Cardinal * num_args;
#endif
#line 80 "XmIcon.w"
{
    if (((XmIconWidget)old)->xmIcon.image != ((XmIconWidget)self)->xmIcon.image) {
	if (! ((XmIconWidget)self)->xmIcon.image) {
	    XtVaSetValues(self, XtNbackgroundPixmap, None, NULL);
	} else {
	    XtVaSetValues
		(self,
		 XtNwidth, ((XmIconWidget)self)->xmIcon.image->attributes.width,
		 XtNheight, ((XmIconWidget)self)->xmIcon.image->attributes.height,
		 XtNbackgroundPixmap, ((XmIconWidget)self)->xmIcon.image->pixmap,
		 NULL);
	    if (XtIsRealized(self))
		XShapeCombineMask(XtDisplay(self), XtWindow(self), ShapeBounding,
				  0, 0, ((XmIconWidget)self)->xmIcon.image->mask, ShapeSet);
	}
    }
    return FALSE;
}
#line 102 "XmIcon.w"
/*ARGSUSED*/
#if NeedFunctionPrototypes
#line 102 "XmIcon.w"
static void realize(Widget self,XtValueMask * mask,XSetWindowAttributes * attributes)
#else
#line 102 "XmIcon.w"
static void realize(self,mask,attributes)Widget self;XtValueMask * mask;XSetWindowAttributes * attributes;
#endif
#line 103 "XmIcon.w"
{
    xmManagerClassRec.core_class.realize(self, mask, attributes);
    XShapeCombineMask(XtDisplay(self), XtWindow(self), ShapeBounding, 0, 0,
		      ((XmIconWidget)self)->xmIcon.image ? ((XmIconWidget)self)->xmIcon.image->mask : None, ShapeSet);
}
#line 112 "XmIcon.w"
/*ARGSUSED*/
#if NeedFunctionPrototypes
#line 112 "XmIcon.w"
static void destroy(Widget self)
#else
#line 112 "XmIcon.w"
static void destroy(self)Widget self;
#endif
#line 113 "XmIcon.w"
{
    if (((XmIconWidget)self)->xmIcon.image) {
	XFreePixmap(XtDisplay(self), ((XmIconWidget)self)->xmIcon.image->pixmap);
	XFreePixmap(XtDisplay(self), ((XmIconWidget)self)->xmIcon.image->mask);
	XpmFreeAttributes(&((XmIconWidget)self)->xmIcon.image->attributes);
    }
}
#line 37 "XmIcon.w"
